### ISSUE TITLE
django-pgtrigger package isn't recognized correctly

### ISSUE THREAD

**Issue by WTK** [fpgmaas/deptry#112]
**Describe the bug**

[django-pgtrigger package](https://pypi.org/project/django-pgtrigger/) package is marked as obsolete by deptry even even though it's actually imported. The problem might be that import name doesn't match the package name (as per their documentation).

**To Reproduce**

1. Install django-pgtrigger
2. In some *.py file add `import pgtrigger`
3. Run `deptry` with default configuration. It will r

**Expected behavior**

deptry should recognize that package pgtrigger is a name of a package exposed by django-pgtrigger installation

**System [please complete the following information]:**

-   OS: e.g. macos 12.4
-   Language Version: 3.9.6
-   Poetry version: none

**Additional context**

None

**Comment by fpgmaas** [comment]
Thanks for raising the issue. 

The problem here is that installing `django-pgtrigger` adds (at least) two directories to the `site-packages` directory in the virtual environment. `django-pgtrigeer` has associated metadata, but `pgtrigger` does not:

```sh
site-packages git:(main) ✗ ls | grep pgtrigger
django_pgtrigger-4.5.2.dist-info
pgtrigger

➜  site-packages git:(main) ✗ tree django_pgtrigger-4.5.2.dist-info 
django_pgtrigger-4.5.2.dist-info
├── INSTALLER
├── LICENSE
├── METADATA
├── RECORD
├── REQUESTED
└── WHEEL

0 directories, 6 files

➜  site-packages git:(main) ✗ tree pgtrigger
pgtrigger
├── __init__.py
├── __pycache__
│   ├── __init__.cpython-39.pyc
│   ├── apps.cpython-39.pyc
│   ├── compiler.cpython-39.pyc
│   ├── contrib.cpython-39.pyc
│   ├── core.cpython-39.pyc
│   ├── features.cpython-39.pyc
│   ├── installation.cpython-39.pyc
│   ├── migrations.cpython-39.pyc
│   ├── models.cpython-39.pyc
│   ├── registry.cpython-39.pyc
│   ├── runtime.cpython-39.pyc
│   ├── utils.cpython-39.pyc
│   └── version.cpython-39.pyc
├── apps.py
├── compiler.py
├── contrib.py
├── core.py
├── features.py
├── installation.py
├── management
│   ├── __init__.py
│   ├── __pycache__
│   │   └── __init__.cpython-39.pyc
│   └── commands
│       ├── __init__.py
│       ├── __pycache__
│       │   ├── __init__.cpython-39.pyc
│       │   └── pgtrigger.cpython-39.pyc
│       └── pgtrigger.py
├── migrations.py
├── models.py
├── registry.py
├── runtime.py
├── utils.py
└── version.py

5 directories, 32 files
```

Therefore, `deptry` will fail to find the associated metadata for `pgtrigger`, and fall back to guessing that the corresponding package name is `pgtrigger`, which in this case is wrong. (A similar thing happens with the `google` packages, but there falling back to `google` is successful, see https://github.com/fpgmaas/deptry/issues/5) 
I will need to investigate exactly why it is that it was decided that `pgtrigger` does not have associated metadata. But I am afraid that in this case, since AFAIK there is no information on the system  linking `django-pgtrigger` to `pgtrigger`, there is no way for `deptry` to make the link between the module and the package, and using `ignore_obsolete` in `pyproject.toml` is the best solution I can offer for now.

If someone has any idea if it's possible to link `django-pgtrigger` to `pgtrigger`, I'm interested to know so we can add another method to link packages and modules if no metadata is found.

**Comment by fpgmaas** [comment]
Some things I want to investigate here, when I have the time:

- Why does `django-pgrtrigger` use the folder structure it currently has in `site-packages`?
- It seems that Poetry (`django-pgtrigger` used Poetry to publish) never adds a `top-level.txt` to the directory. Why? Should we raise an issue for this in Poetry?
- Would having a top-level.txt have sufficed for deptry to recognize this package?

**Comment by fpgmaas** [comment]
1. It's simply caused by the following specification in `pyproject.toml`:

```
[tool.poetry]
name = "django-pgtrigger"
packages = [
  { include = "pgtrigger" }
]
```

2. It seems `top-level.txt` is only added by setuptools, Poetry does not add this.
3. Yes.

I discovered that the metadata file `RECORD` also contains information that we could use. it's description is: `RECORD: records the list of installed files.`.   I will try to build a solution using this.

**Comment by fpgmaas** [comment]
This should now be solved in release [0.5.2](https://github.com/fpgmaas/deptry/releases/tag/0.5.2)
